tests/store/artifact/test_databricks_models_artifact_repo.py

```diff
@@ -166,6 +175,23 @@ def test_list_artifacts(self, databricks_model_artifact_repo):
             assert artifacts[1].file_size is None
             call_endpoint_mock.assert_called_once_with(ANY, REGISTRY_LIST_ARTIFACTS_ENDPOINT)
 
+        # errors from API are propagated through to cli response
+        list_artifact_dir_bad_response_mock = mock.MagicMock()
+        status_code = 404
+        list_artifact_dir_bad_response_mock.status_code = status_code
+        list_artifact_dir_bad_response_mock.text = "{}"
```

Comment:
Could we include an error code and body so that we can be sure that the text does get surfaced as part of the exception?

Reply:
added a response body to ensure it gets printed out. 

tbc though, raise_for_status raises an exception for 4XX and 5XX errors. For a 404 error, it raises an exception along the lines of `404 Client Error: Not Found for url: https://shard-uri/api/2.0/mlflow/model-versions/list-artifacts?name=model&version=1`.

We catch this error and rethrow it in our format: `"API request to list files under path `%s` failed with status code %s. Response body: %s" % (path, response.status_code, response.text)` which should be sufficient because we surface the path to the user and the response code of 404. The response text is usually empty when the request errors out.